File: frontend/src/api/v2ray_server_deploy_stage.ts
import {BaseResponse} from "@/api/base"

export const API_V2RAY_SERVER_DEPLOY_STAGE = "/api/v2ray-server-deploy-stage"

export class V2rayServerDeployStageResponse extends BaseResponse {

  public data = new class {

    public stages = new Array<Stage>()

    public cloudreve = new class {

      public email = ""

      public password = ""
    }
  }
}

export class Stage {

  public stage = 0

  public state = 1
}

export class CloudreveInfo {

  public email = ""

  public password = ""
}

File: controller/v2ray_server_deploy_stage.go
package controller

import (
	"github.com/Luna-CY/v2ray-helper/common/deploy"
	"github.com/Luna-CY/v2ray-helper/common/http/code"
	"github.com/Luna-CY/v2ray-helper/common/http/response"
	"github.com/gin-gonic/gin"
)

func V2rayServerDeployStage(c *gin.Context) {
	stages := deploy.GetManager().GetStage()
	if nil == stages {
		response.Response(c, code.BadRequest, "没有部署中的任务", nil)
	}

	stageList := response.NewEmptyJsonList()
	for _, stage := range stages {
		stageList = append(stageList, gin.H{"stage": stage.Stage, "state": stage.State})
	}

	last := stages[len(stages)-1]
	if deploy.StageConfigGenerate == last.Stage && deploy.StateRunning != last.State {
		deploy.GetManager().Clean()
	}

	response.Success(c, code.OK, &gin.H{"stages": stageList, "cloudreve": gin.H{"email": "admin@cloudreve.org", "password": deploy.GetManager().GetCloudreveAdminPassword()}})
}
